src/lib/logical_query_plan/mock_node.cpp

```diff
@@ -121,6 +97,32 @@ std::string MockNode::description(const DescriptionMode mode) const {
   return stream.str();
 }
 
+std::shared_ptr<UniqueColumnCombinations> MockNode::unique_column_combinations() const {
+  auto unique_column_combinations = std::make_shared<UniqueColumnCombinations>();
+
+  for (const auto& table_key_constraint : _table_key_constraints) {
+    // Discard key constraints that involve pruned column id(s).
+    const auto& key_constraint_column_ids = table_key_constraint.columns();
+    if (std::any_of(_pruned_column_ids.cbegin(), _pruned_column_ids.cend(),
+                    [&key_constraint_column_ids](const auto& pruned_column_id) {
+                      return std::any_of(key_constraint_column_ids.cbegin(), key_constraint_column_ids.cend(),
```

Comment:
Same as above: `find` or `binary_search`?

Reply:
both possible, and I do't think there will be a performance impact on binary search vs find (also, `MockNode` should not be in queries at all).
How about the following (more readable than `find/binary_search(...) != whatever.end()` IMHO)?
```c++
const auto pruned_column_ids = std::unordered_set<ColumnID>{_pruned_column_ids.cbegin(), 
                                                            _pruned_column_ids.cend()};

for (const auto& table_key_constraint : _table_key_constraints) {
  // Discard key constraints that involve pruned column id(s).
  const auto& key_constraint_column_ids = table_key_constraint.columns();
  if (std::any_of(key_constraint_column_ids.cbegin(), key_constraint_column_ids.cend(),
      [&](const auto column_id){ return pruned_column_ids.contains(column_id); })) {
    continue;
  }
  ...
}
```

Reply:
How about using a convenience lambda function to avoid copies? Something like the following:
```
auto contains = [](auto container, auto value) {
  return std::find(container.begin(), container.end(), value) != container.end();
};
```